go/chat/mobilepush.go

```diff
@@ -0,0 +1,182 @@
+}
+func (g *remoteNotificationSuccessHandler) OnConnect(ctx context.Context, conn *rpc.Connection, cli rpc.GenericClient, srv *rpc.Server) error {
+	return nil
+}
+func (g *remoteNotificationSuccessHandler) OnConnectError(err error, reconnectThrottleDuration time.Duration) {
+}
+func (g *remoteNotificationSuccessHandler) OnDisconnected(ctx context.Context, status rpc.DisconnectStatus) {
+}
+func (g *remoteNotificationSuccessHandler) OnDoCommandError(err error, nextTime time.Duration) {}
+func (g *remoteNotificationSuccessHandler) ShouldRetry(name string, err error) bool {
+	return false
+}
+func (g *remoteNotificationSuccessHandler) ShouldRetryOnConnect(err error) bool {
+	return false
+}
+
+type MobilePush struct {
+	globals.Contextified
+	utils.DebugLabeler
+}
+
+func NewMobilePush(g *globals.Context) *MobilePush {
+	return &MobilePush{
+		Contextified: globals.NewContextified(g),
+		DebugLabeler: utils.NewDebugLabeler(g.GetLog(), "MobilePush", false),
+	}
+}
+
+func (h *MobilePush) AckNotificationSuccess(ctx context.Context, pushIDs []string) {
+	defer h.Trace(ctx, func() error { return nil }, "AckNotificationSuccess")()
+	// Get session token
+	nist, _, err := h.G().ActiveDevice.NISTAndUID(ctx)
+	if nist == nil {
+		h.Debug(ctx, "AckNotificationSuccess: got a nil NIST, is the user logged out?")
+		return
+	}
+	if err != nil {
+		h.Debug(ctx, "AckNotificationSuccess: failed to get logged in session: %s", err.Error())
+		return
+	}
+
+	// Make an ad hoc connection to gregor
+	uri, err := rpc.ParseFMPURI(h.G().Env.GetGregorURI())
+	if err != nil {
+		h.Debug(ctx, "AckNotificationSuccess: failed to parse chat server UR: %s", err.Error())
+		return
+	}
+
+	var conn *rpc.Connection
+	if uri.UseTLS() {
+		rawCA := h.G().Env.GetBundledCA(uri.Host)
+		if len(rawCA) == 0 {
+			h.Debug(ctx, "AckNotificationSuccess: failed to parse CAs: %s", err.Error())
+			return
+		}
+		conn = rpc.NewTLSConnection(rpc.NewFixedRemote(uri.HostPort),
+			[]byte(rawCA), libkb.NewContextifiedErrorUnwrapper(h.G().ExternalG()),
+			&remoteNotificationSuccessHandler{}, libkb.NewRPCLogFactory(h.G().ExternalG()),
+			logger.LogOutputWithDepthAdder{Logger: h.G().Log}, rpc.ConnectionOpts{})
+	} else {
+		t := rpc.NewConnectionTransport(uri, nil, libkb.MakeWrapError(h.G().ExternalG()))
+		conn = rpc.NewConnectionWithTransport(&remoteNotificationSuccessHandler{}, t,
+			libkb.NewContextifiedErrorUnwrapper(h.G().ExternalG()),
+			logger.LogOutputWithDepthAdder{Logger: h.G().Log}, rpc.ConnectionOpts{})
+	}
+	defer conn.Shutdown()
+
+	// Make remote successful call on our ad hoc conn
+	cli := chat1.RemoteClient{Cli: NewRemoteClient(h.G(), conn.GetClient())}
+	if err = cli.RemoteNotificationSuccessful(ctx,
+		chat1.RemoteNotificationSuccessfulArg{
+			AuthToken:        gregor1.SessionToken(nist.Token().String()),
+			CompanionPushIDs: pushIDs,
+		}); err != nil {
+		h.Debug(ctx, "AckNotificationSuccess: failed to invoke remote notification success: %",
```

Comment:
`%v", err)`

Reply:
I think this is fixed in my extension PR, its been like this forever,